src/main/java/runge/meme/MemeFrame.java

```diff
@@ -0,0 +1,93 @@
+{
+    MemeList memeList;
+    private final JList<String> nameList;
+    private final DefaultListModel<String> listModel;
+
+    @Inject
+    public MemeFrame(
+            MemeController controller,
+            @Named("listModel") DefaultListModel<String> listModel,
+            @Named("imageLabel") JLabel imageLabel
+    )
+    {
+        this.listModel = listModel;
+        JPanel mainPanel = new JPanel();
+        mainPanel.setLayout(new BorderLayout());
+        setSize(1000, 1000);
+        setTitle("Meme Generator");
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+        setContentPane(mainPanel);
+
+        JPanel textPanel = new JPanel(new BorderLayout());
+        JPanel lowerField = new JPanel(new BorderLayout());
+
+        mainPanel.add(textPanel);
+
+        textPanel.add(lowerField, BorderLayout.SOUTH);
+
+        JTextField upperText = new JTextField("Enter top text here");
+        textPanel.add(upperText, BorderLayout.NORTH);
+
+        JTextField lowerText = new JTextField("Enter lower text here");
+        upperText.setSize(1000, 30);
+        upperText.setFont(new Font(Font.DIALOG, Font.PLAIN, 12));
+        lowerText.setSize(1000, 30);
+        lowerText.setFont(new Font(Font.DIALOG, Font.PLAIN, 12));
+        JLabel lowerLabel = new JLabel();
+        lowerField.add(lowerLabel, BorderLayout.NORTH);
+        JButton updateButton = new JButton("Update");
+        lowerField.add(lowerText, BorderLayout.NORTH);
+        lowerField.add(updateButton, BorderLayout.EAST);
+        imageLabel.setSize(500, 450);
+        imageLabel.setVerticalAlignment(SwingConstants.CENTER);
+        imageLabel.setHorizontalAlignment(SwingConstants.CENTER);
+        textPanel.add(imageLabel, Image.SCALE_DEFAULT);
+
+        nameList = new JList<>(listModel);
+        nameList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        nameList.setLayoutOrientation(JList.VERTICAL);
+
+        JScrollPane listScroller = new JScrollPane(nameList,
+                JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
+                JScrollPane.HORIZONTAL_SCROLLBAR_ALWAYS);
+        listScroller.setPreferredSize(new Dimension(200, 80));
+
+        textPanel.add(listScroller, BorderLayout.EAST);
+        controller.addAndUpdateImage("Drake Hotline Bling");
+
+        MouseListener mouseListener = new MouseAdapter() {
+            public void mouseClicked(MouseEvent e) {
+                if (e.getClickCount() == 2) {
+                    String selectedItem = nameList.getSelectedValue();
+                    controller.addAndUpdateImage(selectedItem);
+                }
+            }
+        };
+        nameList.addMouseListener(mouseListener);
+
+        JButton refresh = new JButton("Refresh");
+        lowerField.add(refresh, BorderLayout.SOUTH);
+
+        refresh.addActionListener(e ->
+        {
+            controller.refreshMainPanel(upperText, lowerText);
+            controller.addAndUpdateImage("Drake Hotline Bling");
```

Comment:
Why are you hardcoding this?

Reply:
that way it automatically sets it back to the first one on the list

Reply:
Unless they change the list and that's no longer the first one...

Reply:
im trying to figure this comment out- and the only way I can see it making sense it so set the image in the frame but I can't call the first element so I don't know what to do